### ISSUE TITLE
Wallaby finds more errors than command-line

### ISSUE THREAD

**Issue by msorens** [wallabyjs/public#1962]
### Issue description or question

Apologies if this is not the right forum...

In a recent issue I reported, there were two errors surfaced by Wallaby that were never surfaced when I ran the same unit test suite on the command line. Yet, AFAIK, both use the same "chrome-headless" setup. Why does Wallaby find more issues? I assume that is actually a good thing but let me ask: is it a good thing? Are these real issues in my code or are they just things that Wallaby gets hung up on for some reason?

### Code editor or IDE name and version

Visual Studio Code v1.30.1

### OS name and version

OSX

**Comment by foxbunny** [comment]
> Why does Wallaby find more issues?

Do you mean you are able to confirm that those issues are legitimate and that the test runner you are using is normally not able to surface them?

**Comment by ArtemGovorov** [comment]
When configured correctly, both Wallaby and command line should report the same results.

If you provide a sample repo where Wallaby reports more errors that CLI, we are happy to investigate and help with the config issue (or, if it is a bug in Wallaby, then urgently fix it).

**Comment by msorens** [comment]
@ArtemGovorov I have seen this a few times over the past year. I reported the latest two examples in https://github.com/wallabyjs/public/issues/1958. One was an issue about the jasmine clock (`Jasmine Clock was unable to install over custom global timer functions.`) and the other was being unable to find a mystery module (`Uncaught Error: Cannot find module '2520'​​ (undefined)`). I fixed the former by adding an `uninstall()` before an `install()` where the clock was being used and the latter by removing the inclusion of one spec file in another. 

But neither of those issues surfaced on the command line.

I can try creating a sample repo with just a couple files to see if the same errors that occurred at scale will still occur in the small, if you need more to go on.

**Comment by msorens** [comment]
@ArtemGovorov Not sure how soon I will be able to find time to build a sample... any comments on the specific error I mentioned above (`Cannot find module... `). Can you confirm whether or not when you have seen that in Wallaby you have also seen it on the command line...?

**Comment by msorens** [comment]
OK, I reduced my working directory to the bare essentials, retaining all the config files and just a couple trivial tests, to reproduce the issue. Here is what wallaby reports in VSCode:
![wallaby in VSCode](https://user-images.githubusercontent.com/6817500/51573217-36b2f380-1e5d-11e9-8570-9011992145ee.png)
Note that I have 2 unit tests: one is reported as passing and the other is not reported at all due to the abort caused by the error I injected (including one spec file in another).
That 5 you see on the "problems" tab, by the way, are just spelling errors reported by a spellchecker.

And here is what I get on the command line from `npm run test`:
![command line test result](https://user-images.githubusercontent.com/6817500/51573256-66fa9200-1e5d-11e9-8cee-3cd59f41f3d0.png)
Both tests reported passing.

I have scrubbed my sample directory but just to be on the safe side I am using a private github repo https://github.com/msorens/wallaby-sample.  Please advise how I can grant you access to it.

**Comment by ArtemGovorov** [comment]
@msorens Please add me (@ArtemGovorov) as a collaborator to the repo.

**Comment by msorens** [comment]
Right, I knew the reason for the wallaby message @ArtemGovorov, but that is not what this issue is about. The question here is: why does wallaby report different results from the command line?  I injected that error specifically as one example of when a difference appears.

**Comment by ArtemGovorov** [comment]
> why does wallaby report different results from the command line?

In this instance it is because Wallaby is trying to be smarter than command line runner and warn you than some of your specs may be executed more than once (one time when a spec file executes and each time when the spec file is being imported from other spec files). 

So in this case, answering your original question in the first comment:

> I assume that is actually a good thing but let me ask: is it a good thing? Are these real issues in my code 

yes, it looks to me like a good thing and yes, you are being warned about real issues in your test code.

**Comment by ArtemGovorov** [comment]
> So is it possible to broadly characterize the types of things that wallaby might be smarter about?

Broadly speaking, Wallaby offers a whole number of features that command line runner does not. To support those features, Wallaby needs to be smarter here and there, however it normally doesn't affect test execution results.

So, I'm struggling to broadly characterize cases when Wallaby reports different results. When configured correctly, **almost always** both Wallaby and command line should report the same results.
